src/test/test-bpf-token.c

```diff
@@ -0,0 +1,21 @@
+/* SPDX-License-Identifier: LGPL-2.1-or-later */
+
+#include <fcntl.h>
+#include <bpf/bpf.h>
+
+#include "tests.h"
+
+int main(void)
+{
+#if __LIBBPF_CURRENT_VERSION_GEQ(1, 5)
+        int bpffs_fd = open("/sys/fs/bpf", O_RDONLY);
+        assert_se(bpffs_fd >= 0);
+
+        int token_fd = bpf_token_create(bpffs_fd, NULL);
```

Comment:
Not a big matter, but this tests no our code... What do you want to test?

At least please split the test case to TEST(bpf_token_create), and use DEFINE_TEST_MAIN() to declare the main function.

Also, maybe better to check capability. Otherwise, if this runs on a container without CAP_BPF, it may fail. Though, I've not checked. So, may not necessary.

Moreover, please test bpf_delegate_commands_to_string() and friends here (by moving them somewhere, maybe in src/shared/bpf-token-util.c or so).

Reply:
This is not a unit test, but a helper used by the real test which is in TEST-07-PID1.
This is why I set it to "manual" so it's not run by meson.

This binary ensures that the bpffs is mounted correctly: calling it with a regular bpffs mount fails.